botcore/utils/extensions.py

```diff
@@ -0,0 +1,52 @@
+"""Utilities for loading discord extensions."""
+
+import importlib
+import inspect
+import pkgutil
+import types
+from typing import NoReturn
+
+
+def unqualify(name: str) -> str:
+    """
+    Return an unqualified name given a qualified module/package `name`.
+
+    Args:
+        name: The module name to unqualify.
+
+    Returns:
+        The unqualified module name.
+    """
+    return name.rsplit(".", maxsplit=1)[-1]
+
+
+def walk_extensions(module: types.ModuleType) -> frozenset[str]:
+    """
+    Yield extension names from the bot.exts subpackage.
```

Comment:
Not accurate since an arbitrary module can be given to search in.

Reply:
https://github.com/python-discord/bot-core/pull/29/commits/94e7367baadb6229cf28530e6c39e06a37e3c9bd